dev_guide/new_app.adoc

```diff
@@ -91,12 +196,97 @@ $ oc new-app ruby-helloworld-sample
 
+.Creating an application from a template in a file
+====
+----
+$ oc new-app -f examples/sample-app/application-template-stibuild.json
+----
+====
+
+==== Template parameters
+
+When creating an application based on a link:templates.html[template],
+use the `-p|--param` argument to set parameter values defined by the template:
+
+.Specifying template parameters with a template
+----
+$ oc new-app ruby-helloworld-sample -p ADMIN_USERNAME=<user>,ADMIN_PASSWORD=<pass>
 ----
-$ oc new-app <template_name> -p ADMIN_USERNAME=<user>,ADMIN_PASSWORD=<pass>
+
+=== Specifying environment variables
+
+When generating applications from 
+link:#specifying-source-code[source] or an
+link:#specifying-an-image[image], you can use the `-e|--env` argument to specify environment
+to be passed to the application container at run time.
+
+.Setting environment variables when creating an app for a database image
+----
+$ oc new-app openshift/postgresql-92-centos7 -e POSTGRESQL_USER=user -e POSTGRESQL_DATABASE=db -e POSTGRESQL_PASSWORD=pass
+----
+
+
+=== Specifying labels
+
+When generating applications from
+link:#specifying-source-code[source],
+link:#specifying-an-image[images], or 
+link:#specifying-a-templates[templates], you can use the `l|--label` argument to add labels to objects created by new-app.
+This is recommended, as labels make it easy to collectively select, manipulate, and delete objects
+associated with the application.
+
+.Using the label argument to label objects created by new-app
+====
+----
+$ oc new-app https://github.com/openshift/ruby-hello-world -l name=hello-world
 ----
+====
+
+=== Multiple components and grouping
+
+New-app allows creating multiple applications from 
+link:#specifying-source-code[source], 
+link:#specifying-an-image[images], or 
+link:#specifying-templates[templates] at once. 
+To do this, simply specify multiple parameters to the new-app call. Labels 
+specified in the commannd line will apply to all objects created by the one call. 
+Environment variables will apply to all components created from source or images.
+
+.Creating an application from a source repository and a DockerHub image
+====
+----
+$ oc new-app https://github.com/openshift/ruby-hello-world mysql
+----
+====
+
+[NOTE]
+====
+If a source code repository and a builder image are specified as separate arguments, new-app will use the builder image as the builder
+for the source code repository. If this is not the intent, simply specify a specific builder image for the source using the `~` separator.
+====
+
+==== Grouping images and source in a single pod
+New-app allows deploying multiple images together in a single pod. In order to specify which images to group together, use the `+` separator. The 
+`--group` command line argument can also be used to specify which images should be grouped together. To group the image built from a source repository
+with other images, specify its builder image in the group.
+
+.Deploying two images in a single pod
```

Comment:
lump these last couple sections under an "advanced usage" section, and add a mention of using "-o" to just output json/yaml instead of creating things (so you can edit the content before creating the objects)

also i don't think you mentioned the --code flag.. is there a circumstance under which that would be necessary?